llvm/lib/Transforms/IPO/RepoPruning.cpp

```diff
@@ -123,14 +123,125 @@ repodefinition::DigestType toDigestType(pstore::index::digest D) {
   return Digest;
 }
 
+static void
+collectSectionXFixupName(pstore::database const &Db,
+                         pstore::repo::generic_section const &Section,
+                         StringSet<> &XFixupNames) {
+  for (auto Xfx : Section.xfixups()) {
+    LLVM_DEBUG(dbgs() << "    Adding XFixup name: "
+                      << pstore::indirect_string::read(Db, Xfx.name).to_string()
+                      << '\n');
+    XFixupNames.insert(
+        std::move(pstore::indirect_string::read(Db, Xfx.name).to_string()));
+
+  }
+}
+
+#define PSTORE_GENERIC_SECTION_KINDS                                           \
```

Comment:
Rather than replicating that knowledge here, this code should really be using `pstore::repo::enum_to_section<>` to map from the section-type to the types used to store the associated section data.

Reply:
Good point, thanks. I have fixed it following your advice.